apps/financial-aid/backend/src/app/modules/personalTaxReturn/personalTaxReturn.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@nestjs/common'
+import { Base64 } from 'js-base64'
+import { PersonalTaxReturnApi } from '@island.is/clients/rsk/personal-tax-return'
+import { FileService } from '../file'
+import fetch from 'isomorphic-fetch'
+
+@Injectable()
+export class PersonalTaxReturnService {
+  constructor(
+    private personalTaxReturnApi: PersonalTaxReturnApi,
+    private fileService: FileService,
+  ) {}
+
+  private async callPersonalTaxReturn(nationalId: string, year: number) {
+    return await this.personalTaxReturnApi
+      .personalTaxReturnInPdf(nationalId, year.toString())
+      .catch(() => {
+        return { success: false, errorText: '', content: '' }
+      })
+  }
+
+  async personalTaxReturn(nationalId: string, folder: string) {
+    let year = new Date().getFullYear() - 1
```

Comment:
This varibles mutate, so it could be last year og two years ago.  

The reason for this is the tax api doesn't fetch latest tax return but only by year. So if the call fails then we try the year before that if that fails then we allow the user to upload the tax return for themself.